base/uk.ac.stfc.isis.ibex.scriptgenerator/src/uk/ac/stfc/isis/ibex/scriptgenerator/ScriptGeneratorSingleton.java

```diff
@@ -390,4 +397,69 @@ public void refreshParameterValidityChecking() {
 			threadError = true;
 		}
 	}
+	
+	/**
+	 * Generate a script and save it to file.
+	 * 
+	 * @return The filepath the script is saved to.
+	 * @throws InvalidParamsException If the parameters are invalid a script cannot be generated.
+	 * @throws IOException If we fail to create and write to a file.
+	 */
+	public String generate() throws InvalidParamsException, IOException {
+		return generateTo("C:/Scripts/");
```

Comment:
Can we get this file path from the preferences plugin so that it can be easily adjusted in built clients later if needed.

In particular, I expect we may be asked to save scripts to the `U:\` drive rather than the `C:\` drive